torchao/quantization/quant_primitives.py

```diff
@@ -69,6 +72,41 @@ class ZeroPointDomain(Enum):
     })
 
 
+# def register_custom_op(name: str):
+#     from torch._inductor.decomposition import register_decomposition
+
+#     def decorator(fn):
+#         if TORCH_VERSION_AFTER_2_5:
+#             opdef = torch.library.custom_op(name, mutates_args=())(fn)
+#             opdef.register_fake(fn)
+#             register_decomposition([opdef._opoverload])(fn)
+#             return opdef
+#         else:
+#             return fn
+
+#     return decorator
+
+quant_lib = torch.library.Library("quant", "FRAGMENT")
+
+def register_custom_op(lib, schema: str):
+    from torch._inductor.decomposition import register_decomposition
+
+    def decorator(fn):
+        if TORCH_VERSION_AFTER_2_5:
+            # TODO: change order
+            lib_namespace = lib.ns
+            op_name = schema.split("(")[0]
```

Comment:
Maybe construct schema object from string and query op name? I thought such a functionality existed, but not sure

Reply:
oh, not sure if this is possible, cc @zou3519 is there a better way to get `op_name` here?

Reply:
torch._C.parse_schema will give you a FunctionSchema object

Reply:
I just used `fn.__name__[1:]` for now